lang/test/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransformTest.kt

```diff
@@ -5737,6 +5737,21 @@ class StaticTypeInferenceVisitorTransformTest : VisitorTransformTestBase() {
                     )
                 )
             ),
+            TestCase(
+                name = "function signature without optional/variadic arguments, null propagating given non matching type for a path",
+                originalSql = "UPPER(x.y)",
+                globals = mapOf("x" to StructType(mapOf("y" to INT))),
+                handler = expectSemanticProblems(
+                    expectedProblems = listOf(
+                        createInvalidArgumentTypeForFunctionError(
+                            sourceLocation = SourceLocationMeta(1L, 7L, 1L),
+                            functionName = "upper",
+                            expectedArgType = unionOf(STRING, SYMBOL),
+                            actualType = INT
+                        )
+                    )
+                )
+            ),
```

Comment:
Could you also add the similar path test case brought up by the customer involving list references of the wrong type:

Something like `UPPER(x[0])` with a globals of `"x" to ListType(INT)`?

Reply:
Will add the test case in the next commit.